network/msgqueue/message_queue.go

```diff
@@ -20,18 +24,22 @@ type messageContainer struct {
 // To solve this issue we have a message broker from which the instance pulls new messages, this also reduces concurrency issues as the instance is now single threaded.
 // The message queue has internal logic to organize messages by their round.
 type MessageQueue struct {
-	msgMutex    sync.Mutex
+	msgMutex    sync.RWMutex
 	indexFuncs  []IndexFunc
-	queue       map[string][]*messageContainer // = map[index][messageContainer.id]messageContainer
-	allMessages map[string]*messageContainer
+	queue       map[string][]messageContainer // = map[index][messageContainer.id]messageContainer
```

Comment:
@nivBlox not sure this is relevant to this PR.. might be a good idea to separate

Reply:
@amir-blox is cleaning this area

Reply:
added in https://github.com/bloxapp/ssv/pull/214